### ISSUE TITLE
Patch Error: UnicodeDecodeError

### ISSUE THREAD

**Issue by psadi** [daylinmorgan/monolisa-nerdfont-patch#7]
Im trying to patch the font and facing the below error, any advice ?

<img width="1548" alt="image" src="https://github.com/daylinmorgan/monolisa-nerdfont-patch/assets/22428691/5a89cf0c-33f4-464c-ba48-aedf629305e3">

**Comment by psadi** [comment]
Not sure what kind of bug this is, but when re-ran with verbose the script exits, correctly and im able to get the patched font

https://github.com/daylinmorgan/monolisa-nerdfont-patch/assets/22428691/0907ce92-f0c7-45e6-871f-114f972faf8c

**Comment by daylinmorgan** [comment]
@psadi could you verify if the recent commit runs successfully on your machine without `--verbose`? 

Also, just to make sure you know, with the current command you are only adding three symbol sets, see `font-patcher --help` for a list of the symbols and their corresponding flags or use `-c/--complete` to add all of them.

**Comment by psadi** [comment]
The issue still remains,

**Without Verbose & With -c**
<img width="1716" alt="image" src="https://github.com/daylinmorgan/monolisa-nerdfont-patch/assets/22428691/da275dd1-a2a3-440b-a655-0cfafc66075f">

**With Verbose & With -c:**
<img width="1792" alt="image" src="https://github.com/daylinmorgan/monolisa-nerdfont-patch/assets/22428691/cff494d8-9742-42ee-a981-e98fcdbcdb5b">

<img width="1792" alt="image" src="https://github.com/daylinmorgan/monolisa-nerdfont-patch/assets/22428691/3993a4dd-4db0-4ad3-98cb-0fe83cbc369b">

**Comment by daylinmorgan** [comment]
Yay encoding errors.... docker or your system is returning text with a different encoding. I think I can catch this at runtime.

What is the output of the following command: 
```sh
python -c 'import locale; print(locale.getpreferredencoding())'
```

**Comment by psadi** [comment]
I just tested the patched font locally with `encoding=None`, and it seems to be working correctly!

However, I'm not sure if there might be any unintended side effects from this change

<img width="1792" alt="image" src="https://github.com/daylinmorgan/monolisa-nerdfont-patch/assets/22428691/f5cadccc-6096-4719-a649-8f29ea83a5c6">

Would it be okay to submit a pull request with this fix, or should we consider adding it directly?

**Comment by psadi** [comment]
> Yay encoding errors.... docker or your system is returning text with a different encoding. I think I can catch this at runtime.
> 
> What is the output of the following command:
> 
> ```shell
> python -c 'import locale; print(locale.getpreferredencoding())'
> ```

Please find output

<img width="788" alt="image" src="https://github.com/daylinmorgan/monolisa-nerdfont-patch/assets/22428691/6af8de79-0ab8-4409-bf08-d3345fa926aa">

**Comment by daylinmorgan** [comment]
Removing the encoding all together will be an issue if there is an error in the subprocess because then the stdout/stderr is returned to the user: https://github.com/daylinmorgan/monolisa-nerdfont-patch/blob/6f5ec7e73e781a11cd37dceb698a9ad3bd8777ee/patch-monolisa#L97-L105

I don't know why the output of `docker run` seems to not be UTF-8

Could you try running it after setting the encoding to "ISO-8859-1"?

I noticed your python was installed by `homebrew` are you using OS X? 

Really the simplest solution for this script is probably to just [ignore encoding errors](https://docs.python.org/3/library/codecs.html#error-handlers), by setting `errors='replace'` in the `subprocess.run`

**Comment by psadi** [comment]
Correct, I'm currently on OSX. Wouldn't it be better to enclose the subprocess block in `try..except` and catch explicit errors.

I was thinking something like this so `None` would be in use for unsupported encodings and still catch errors for a graceful exit.

```python
import locale

try:
  p = subprocess.run(
      command,
      stdout=None if verbose else subprocess.PIPE,
      stderr=None if verbose else subprocess.STDOUT,
      encoding = "utf-8" if locale.getpreferredencoding().lower() == 'utf-8' else None
  )
except (CalledProcessError, UnicodeDecodeError, ValueEror) as err:
  # do something

```

**Comment by psadi** [comment]
> Removing the encoding all together will be an issue if there is an error in the subprocess because then the stdout/stderr is returned to the user:
> 
> https://github.com/daylinmorgan/monolisa-nerdfont-patch/blob/6f5ec7e73e781a11cd37dceb698a9ad3bd8777ee/patch-monolisa#L97-L105
> 
> I don't know why the output of `docker run` seems to not be UTF-8
> 
> Could you try running it after setting the encoding to "ISO-8859-1"?
> 
> I noticed your python was installed by `homebrew` are you using OS X?
> 
> Really the simplest solution for this script is probably to just [ignore encoding errors](https://docs.python.org/3/library/codecs.html#error-handlers), by setting `errors='replace'` in the `subprocess.run`

A noteworthy point: I faced the same error using `make patch`  as well

**Comment by daylinmorgan** [comment]
> Wouldn't it be better to enclose the subprocess block in `try..except` and catch explicit errors.

Not really because we don't actually need the output for anything so might as well lazily decode it and move on. Also `subproccess.run` doesn't raise a `CalledProccessError` without `check = True`.

Also, as we've learned this line `encoding = "utf-8" if locale.getpreferredencoding().lower() == 'utf-8' else None` on your machine would set the encoding to utf-8 which it's not.

> A noteworthy point: I faced the same error using make patch as well

Ya `make patch` doesn't do anything special.

**Comment by psadi** [comment]
Agreed, so we could just catch `UnicodeDecodeError` and re-run as `None` ?

<img width="795" alt="image" src="https://github.com/daylinmorgan/monolisa-nerdfont-patch/assets/22428691/12a45157-212c-4591-a6c8-1216a2e6b9b1">

**Comment by daylinmorgan** [comment]
We don't want to run the subprocess twice because the error has nothing to do with it. And this doesn't solve the problem I mentioned [here](https://github.com/daylinmorgan/monolisa-nerdfont-patch/issues/7#issuecomment-1979368996) that we still want to decode the result to print in the case of errors 

You have `fontforge` installed? Is there a reason you are using the `docker ` mode then? 

What is the output with `make patch` or `patch-monolisa -f Monolisa -c` if you are still getting encoding errors then I think this has something to with your environment of locale settings.
